projects/plugins/jetpack/class.jetpack.php

```diff
@@ -3873,52 +3881,61 @@ public function deactivate_dialog( $hook ) {
 			&& self::is_connection_ready()
 		) {
 
-			$active_plugins_using_connection = Connection_Plugin_Storage::get_all();
-
-			if ( count( $active_plugins_using_connection ) > 1 ) {
+			// Register jp-tracks-functions dependency.
+			Tracking::register_tracks_functions_scripts( true );
 
-				add_thickbox();
+			// add a deactivation script that will pick up deactivation actions for the Jetpack plugin.
+			$script_deps_path    = JETPACK__PLUGIN_DIR . '_inc/build/plugins-page.asset.php';
+			$script_dependencies = array( 'wp-polyfill' );
+			if ( file_exists( $script_deps_path ) ) {
+				$asset_manifest      = include $script_deps_path;
+				$script_dependencies = $asset_manifest['dependencies'];
+			}
 
-				// Register jp-tracks-functions dependency.
-				Tracking::register_tracks_functions_scripts();
+			wp_enqueue_script(
+				'jetpack-plugins-page-js',
+				plugins_url( '_inc/build/plugins-page.js', JETPACK__PLUGIN_FILE ),
+				$script_dependencies,
+				JETPACK__VERSION,
+				true
+			);
 
-				wp_enqueue_script(
-					'jetpack-deactivate-dialog-js',
-					Assets::get_file_url_for_environment(
-						'_inc/build/jetpack-deactivate-dialog.min.js',
-						'_inc/jetpack-deactivate-dialog.js'
-					),
-					array( 'jquery', 'jp-tracks-functions' ),
-					JETPACK__VERSION,
-					true
-				);
+			// load this on pages where needed.
+			wp_localize_script(
+				'jetpack-plugins-page-js',
+				'deactivate_dialog',
+				array(
+					'title'            => __( 'Deactivate Jetpack', 'jetpack' ),
+					'deactivate_label' => __( 'Disconnect and Deactivate', 'jetpack' ),
+				)
+			);
 
-				wp_localize_script(
-					'jetpack-deactivate-dialog-js',
-					'deactivate_dialog',
-					array(
-						'title'            => __( 'Deactivate Jetpack', 'jetpack' ),
-						'deactivate_label' => __( 'Disconnect and Deactivate', 'jetpack' ),
-						'tracksUserData'   => Jetpack_Tracks_Client::get_connected_user_tracks_identity(),
-					)
-				);
+			wp_set_script_translations( 'jetpack-plugins-page-js', 'jetpack' );
 
-				add_action( 'admin_footer', array( $this, 'deactivate_dialog_content' ) );
+			// Add objects to be passed to the initial state of the app.
+			// Use wp_add_inline_script instead of wp_localize_script, see https://core.trac.wordpress.org/ticket/25280.
+			wp_add_inline_script( 'jetpack-plugins-page-js', 'var Initial_State=JSON.parse(decodeURIComponent("' . rawurlencode( wp_json_encode( Jetpack_Redux_State_Helper::get_initial_state() ) ) . '"));', 'before' );
 
-				wp_enqueue_style( 'jetpack-deactivate-dialog', plugins_url( 'css/jetpack-deactivate-dialog.css', JETPACK__PLUGIN_FILE ), array(), JETPACK__VERSION );
-			}
+			// Include styles for WP Components ( Includes the modal used for disconnection ).
+			wp_enqueue_style( 'wp-components' );
```

Comment:
Isn't this something we could declare as a dependency of `jetpack-plugins-page-css` instead?

Reply:
Oh yeah, good idea!